From c31d1b2ac31d70fc81cc2aafc3a0320456107348 Mon Sep 17 00:00:00 2001
From: Terry Zhou <bjzhou@marvell.com>
Date: Thu, 5 May 2016 10:47:49 +0800
Subject: [PATCH 0211/1345] ARM: mvebu: change place of coredivclk in Armada
 370/XP DT structure

commit  6d670e749c6675a3de6126273abf33d86da03fa3 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

'coredivclk' in Msys SoC's is not compatible with Armada 370/XP.
Because all those machines will share common armada-370-xp.dtsi
file, above definition have to be placed in a different place of DT
structure.

This commit moves 'coredivclk' to both armada-xp.dtsi and armada-370.dtsi.
It is a preparation for adding Msys SoC kernel support.

Change-Id: I3125586e60126a516a7ab9930ce0f18f2addf024
Signed-off-by: Terry Zhou <bjzhou@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29692
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Reviewed-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-370-xp.dtsi |    8 --------
 arch/arm/boot/dts/armada-xp.dtsi     |    8 ++++++++
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/armada-370-xp.dtsi b/arch/arm/boot/dts/armada-370-xp.dtsi
index 09495e8..383d92a 100644
--- a/arch/arm/boot/dts/armada-370-xp.dtsi
+++ b/arch/arm/boot/dts/armada-370-xp.dtsi
@@ -190,14 +190,6 @@
 				reg = <0x18000 0x38>;
 			};
 
-			coredivclk: corediv-clock@18740 {
-				compatible = "marvell,armada-370-corediv-clock";
-				reg = <0x18740 0xc>;
-				#clock-cells = <1>;
-				clocks = <&mainpll>;
-				clock-output-names = "nand";
-			};
-
 			mbusc: mbus-controller@20000 {
 				compatible = "marvell,mbus-controller";
 				reg = <0x20000 0x100>, <0x20180 0x20>,
diff --git a/arch/arm/boot/dts/armada-xp.dtsi b/arch/arm/boot/dts/armada-xp.dtsi
index fa1e881..e526b27 100644
--- a/arch/arm/boot/dts/armada-xp.dtsi
+++ b/arch/arm/boot/dts/armada-xp.dtsi
@@ -143,6 +143,14 @@
 				clocks = <&coreclk 1>;
 			};
 
+			coredivclk: corediv-clock@18740 {
+				compatible = "marvell,armada-370-corediv-clock";
+				reg = <0x18740 0xc>;
+				#clock-cells = <1>;
+				clocks = <&mainpll>;
+				clock-output-names = "nand";
+			};
+
 			cpu-config@21000 {
 				compatible = "marvell,armada-xp-cpu-config";
 				reg = <0x21000 0x8>;
-- 
1.7.9.5

